analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -475,4 +476,29 @@ class SpdxDocumentFile(
 
         return listOf(ProjectAnalyzerResult(project, packages.toSortedSet()))
     }
+
+    /**
+     * Create the final [PackageManagerResult] by making sure that packages are removed from [projectResults] that
+     * are also referenced as project dependencies.
+     */
+    override fun createPackageManagerResult(
+        projectResults: Map<File, List<ProjectAnalyzerResult>>
+    ): PackageManagerResult {
+        val projectIds =
+            projectResults.flatMapTo(mutableSetOf()) { projResult -> projResult.value.map { it.project.id } }
+
+        val filteredResults = projectResults.mapValues { entry ->
+            entry.value.map { projectResult ->
+                val projectReferences = projectResult.packages.filterTo(mutableSetOf()) { it.id in projectIds }
+                projectResult.takeIf { projectReferences.isEmpty() }
+                    ?: projectResult.copy(packages = (projectResult.packages - projectReferences).toSortedSet())
+                        .also {
+                            log.info { "Removing ${projectReferences.size} packages that are projects." }
+                            log.debug { projectReferences.joinToString { it.id.toCoordinates() } }
```

Comment:
IMO this could be merged into one debug statement, but I'm also fine if you want to keep it as is.

Reply:
I would prefer keeping it this way. With out default log level being INFO, this makes sure that we get some hint about a potential problem in SPDX files, but the log is not too verbose.